src/OpenLoco/src/GameCommands/Airports/RemoveAirport.cpp

```diff
@@ -0,0 +1,275 @@
+        {
+            const auto portPos = World::Pos3(searchTile.pos + pos, pos.z);
+
+            if ((flags & Flags::aiAllocated) != 0 && (flags & Flags::apply) != 0)
+            {
+                auto tile = World::TileManager::get(portPos);
+                auto* surfaceEl = tile.surface();
+                if (surfaceEl != nullptr)
+                {
+                    surfaceEl->setAiAllocated(false);
+                }
+            }
+
+            if ((flags & Flags::apply) == 0)
+            {
+                continue;
+            }
+
+            auto* stationEl = getStationEl(portPos);
+            if (stationEl == nullptr)
+            {
+                return false;
+            }
+
+            if ((flags & (Flags::aiAllocated)) == 0)
+            {
+                Ui::ViewportManager::invalidate(World::Pos2(portPos), stationEl->baseHeight(), stationEl->clearHeight(), ZoomLevel::eighth);
+            }
+
+            World::TileManager::removeElement(*reinterpret_cast<World::TileElement*>(stationEl));
+        }
+
+        return true;
+    }
+
+    // 0x00493519
+    static bool isAirportInUseByVehicle(const StationId stationId)
+    {
+        auto vehicleList = VehicleManager::VehicleList();
+        for (auto* vehicle : vehicleList)
+        {
+            if (vehicle->vehicleType != VehicleType::aircraft)
+                continue;
+
+            if (vehicle->tileX == -1)
+                continue;
+
+            if (vehicle->stationId == stationId)
+            {
+                GameCommands::setErrorText(StringIds::currently_in_use_by_at_least_one_vehicle);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    // 0x0049372F
+    static currency32_t loc_49372F(const StationId stationId, const World::StationElement& stationEl, const World::Pos3 pos, const uint8_t flags)
+    {
+        const auto rotation = stationEl.rotation();
+        const auto objectId = stationEl.objectId();
+
+        if (stationId != StationId::null)
+        {
+            if (isAirportInUseByVehicle(stationId))
+            {
+                return FAILURE;
+            }
+        }
+
+        const auto* airportObj = ObjectManager::get<AirportObject>(objectId);
+
+        // Calculate base removal cost
+        currency32_t totalCost = Economy::getInflationAdjustedCost(airportObj->sellCostFactor, airportObj->costIndex, 6);
+
+        // Adjust number of airports for nearest town
+        auto maybeTown = TownManager::getClosestTownAndDensity(pos);
+        if (maybeTown && (flags & Flags::apply) != 0)
+        {
+            if ((flags & (Flags::aiAllocated | Flags::ghost)) == 0)
+            {
+                auto* town = TownManager::get(maybeTown->first);
+                town->numberOfAirports--;
+            }
+        }
+
+        // TODO: introduce struct
+        auto* var_9C = reinterpret_cast<const int8_t*>(airportObj->var_9C);
+        while (var_9C[0] != -1)
```

Comment:
instead of this we should make an accessor function on airport that returns a span. See for example `BuildingObject::getBuildingParts` which is also a 0xFF terminated list.